<launch>


  <!-- system -->
  <arg name="control_rate" default="30.0" doc="controller execution rate [Hz]" />

  <!-- enable flags -->
  <arg name="show_debug_info" default="false" />
  <arg name="enable_smooth_stop" default="true" />
  <arg name="enable_pub_debug" default="true" />

  <arg name="velocity_controller_param_path" default="$(find-pkg-share velocity_controller)/config/velocity_controller_param.yaml" />

  <node pkg="velocity_controller" exec="velocity_controller" name="velocity_controller" output="screen">
    <param from="$(var velocity_controller_param_path)"/>

    <remap from="current_velocity" to="/localization/twist"/>
    <remap from="control_cmd" to="longitudinal/control_cmd"/>
    <remap from="current_trajectory" to="/planning/scenario_planning/trajectory"/>

    <!-- system -->
    <param name="control_rate" value="$(var control_rate)"/>

    <!-- enable flags -->
    <param name="show_debug_info" value="$(var show_debug_info)" />
    <param name="enable_smooth_stop" value="$(var enable_smooth_stop)" />
    <param name="enable_pub_debug" value="$(var enable_pub_debug)" />


  </node>


</launch>
